From 1bd9be7d24cd4d576a204f5985cfa2dfde780c41 Mon Sep 17 00:00:00 2001
From: Rafael Stahl <r.stahl@tum.de>
Date: Wed, 27 Jan 2021 14:07:07 +0100
Subject: [PATCH 2/3] micro config

---
 cmake/config.cmake | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/cmake/config.cmake b/cmake/config.cmake
index 7b29df648..40cc5b4eb 100644
--- a/cmake/config.cmake
+++ b/cmake/config.cmake
@@ -85,7 +85,7 @@ set(USE_VULKAN OFF)
 set(USE_OPENGL OFF)
 
 # Whether enable MicroTVM runtime
-set(USE_MICRO OFF)
+set(USE_MICRO ON)
 
 # Whether enable RPC runtime
 set(USE_RPC ON)
@@ -106,7 +106,7 @@ set(USE_GRAPH_EXECUTOR_CUDA_GRAPH OFF)
 set(USE_PROFILER ON)
 
 # Whether enable uTVM standalone runtime
-set(USE_MICRO_STANDALONE_RUNTIME OFF)
+set(USE_MICRO_STANDALONE_RUNTIME ON)
 
 # Whether build with LLVM support
 # Requires LLVM version >= 4.0
-- 
2.25.1

