From 2445dd657a008fe89cb16104442ca228f74e3c0c Mon Sep 17 00:00:00 2001
From: Rafael Stahl <r.stahl@tum.de>
Date: Wed, 27 Jan 2021 12:16:08 +0100
Subject: [PATCH 2/2] utvm build fix

---
 python/tvm/micro/build.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/python/tvm/micro/build.py b/python/tvm/micro/build.py
index cad385b9b..9b9f0e7fa 100644
--- a/python/tvm/micro/build.py
+++ b/python/tvm/micro/build.py
@@ -52,7 +52,7 @@ class Workspace:
 
 
 # Required C runtime libraries, in link order.
-CRT_RUNTIME_LIB_NAMES = ["utvm_rpc_server", "utvm_rpc_common", "common"]
+CRT_RUNTIME_LIB_NAMES = ["utvm_rpc_server", "utvm_rpc_common", "common", "memory"]
 
 
 TVM_ROOT_DIR = os.path.realpath(os.path.join(os.path.dirname(__file__), "..", "..", ".."))
-- 
2.17.1

