From 78f260975c8c54eef020d170dd0d728bffc260bb Mon Sep 17 00:00:00 2001
From: Rafael Stahl <r.stahl@tum.de>
Date: Sun, 14 Mar 2021 15:23:01 +0100
Subject: [PATCH] c codegen: support tir.round

---
 src/target/source/codegen_c.cc | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/target/source/codegen_c.cc b/src/target/source/codegen_c.cc
index af175c7f2..9da03f4d6 100644
--- a/src/target/source/codegen_c.cc
+++ b/src/target/source/codegen_c.cc
@@ -648,6 +648,10 @@ void CodeGenC::VisitExpr_(const CallNode* op, std::ostream& os) {  // NOLINT(*)
       os << " != ";
       this->PrintExpr(op->args[0], os);
       os << ")";
+    } else if (ptr_op->name == "tir.round") {
+      os << "(";
+      this->PrintExpr(op->args[0], os);
+      os << " + 0.5f)";
     } else {
       LOG(FATAL) << "Unresolved call " << op->op;
     }
-- 
2.17.1

