diff --git a/python/tvm/relay/frontend/tflite.py b/python/tvm/relay/frontend/tflite.py
index d49340b85..dce369b5d 100644
--- a/python/tvm/relay/frontend/tflite.py
+++ b/python/tvm/relay/frontend/tflite.py
@@ -1226,7 +1226,9 @@ class OperatorConverter(object):
         lhs_tensor = input_tensors[0]
         rhs_tensor = input_tensors[1]
         lhs_expr = self.get_tensor_expr(lhs_tensor)
+        lhs_expr_ = relay.annotation.compiler_begin(lhs_expr, "ccompiler")
         rhs_expr = self.get_tensor_expr(rhs_tensor)
+        rhs_expr_ = relay.annotation.compiler_begin(rhs_expr, "ccompiler")
 
         output_tensors = self.get_output_tensors(op)
         assert len(output_tensors) == 1, "output tensors length should be 1"
@@ -1256,7 +1258,7 @@ class OperatorConverter(object):
                 output_zero_point=output_tensor.qnn_params["zero_point"],
             )
         else:
-            out = relay_op(lhs_expr, rhs_expr)
+            out = relay_op(lhs_expr_, rhs_expr_)
 
         # Options (fused_activation_function)
         options = None
@@ -1288,7 +1290,8 @@ class OperatorConverter(object):
                 )
             else:
                 out = self.convert_fused_activation_function(out, fused_activation_fn)
-        return out
+        out_ = relay.annotation.compiler_end(out, "ccompiler")
+        return out_
 
     def convert_add(self, op):
         """Convert TFLite ADD"""
